fe/fe-core/src/main/java/org/apache/doris/datasource/hive/PooledHiveMetaStoreClient.java

```diff
@@ -53,7 +53,8 @@ public class PooledHiveMetaStoreClient {
     private static final Logger LOG = LogManager.getLogger(PooledHiveMetaStoreClient.class);
 
     private static final HiveMetaHookLoader DUMMY_HOOK_LOADER = t -> null;
-    private static final short MAX_LIST_PARTITION_NUM = 10000;
+    // -1 means no limit on the partitions returned.
+    private static final short MAX_LIST_PARTITION_NUM = -1;
```

Comment:
better change it to a mutable FE config.
I am afraid if there are too many partitions, it will causing FE OOM.
Set as a config, and default value can be -1, if encounter error, we can set it manually.